trino/trino.go

```diff
@@ -595,6 +601,23 @@ func (st *driverStmt) CheckNamedValue(arg *driver.NamedValue) error {
 	if reflect.TypeOf(arg.Value).Kind() == reflect.Slice {
 		return nil
 	}
+
+	// check callback
+	if arg.Name == trinoProgressCallbackParam {
+		callback, ok := arg.Value.(QueryProgressCallback)
+		if ok {
+			st.conn.callback = callback
```

Comment:
Why do this both here and in `exec()`? I think this method should only do validation and not modify any properties of the connection.

Reply:
Oops, my bad. I forgot to remove it at exec() part because `CheckNamedValue` is called by sql.

Reply:
It should be done in exec, even if just for consistency with `X-Trino-User`.

Reply:
Will fix it as you said. 
Also, what do you think about adding default period if user does not give period header? 1000ms?

Reply:
I don't think we can find a good default value, so let's just make it required.

Reply:
Ah, I forgot that only primitive types are supported as query params by default, so if a callback is passed we'd still need to check for it here. But I'd still separate validation from the assignment.

Reply:
I can't seem to find appropriate location for required header, please let me know.

For now, I put it at exec() function, after all headers.
![image](https://user-images.githubusercontent.com/25147023/182127435-8da7d0df-e65c-4f18-bc6c-3f6f6e209760.png)